From: Gal Pressman <galp@mellanox.com>
Date: Mon, 27 Jun 2016 12:08:35 +0300
Subject: [PATCH] net/mlx5e: Check for BlueFlame capability before allocating
 SQ uar
Patch-mainline: v4.7-rc6
Git-commit: fd4782c21359cfd52af4c3180a2bb6bad55c1eba
References: bsc#966170 FATE#320225 bsc#966172 FATE#320226

Previous to this patch mapping was always set to write combining without
checking whether BlueFlame is supported in the device.

Fixes: 0ba422410bbf ('net/mlx5: Fix global UAR mapping')
Signed-off-by: Gal Pressman <galp@mellanox.com>
Signed-off-by: Saeed Mahameed <saeedm@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/net/ethernet/mellanox/mlx5/core/en_main.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/ethernet/mellanox/mlx5/core/en_main.c
+++ b/drivers/net/ethernet/mellanox/mlx5/core/en_main.c
@@ -541,7 +541,7 @@ static int mlx5e_create_sq(struct mlx5e_
 	int txq_ix;
 	int err;
 
-	err = mlx5_alloc_map_uar(mdev, &sq->uar, true);
+	err = mlx5_alloc_map_uar(mdev, &sq->uar, !!MLX5_CAP_GEN(mdev, bf));
 	if (err)
 		return err;
 
